packages/core/app-layout/src/components/pageHeader/AppPageHeader.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <div class="kong-ui-app-page-header">
+    <div
+      v-if="hasBreadcrumbs"
+      class="page-header-breadcrumbs"
+      data-testid="page-header-breadcrumbs"
+    >
+      <KBreadcrumbs :items="breadcrumbs">
+        <template #divider>
+          <span class="page-header-breadcrumb-divider">/</span>
+        </template>
+      </KBreadcrumbs>
+    </div>
+
+    <div class="page-header-title-section">
+      <div class="page-header-title-wrapper">
+        <div
+          v-if="$slots.icon"
+          class="page-header-title-icon"
+          data-testid="page-header-title-icon"
+        >
+          <slot name="icon" />
+        </div>
+        <h1
+          class="page-header-title"
+          data-testid="page-header-title"
+          :title="title"
+        >
+          {{ title }}
+        </h1>
+        <div
+          v-if="$slots.badge"
+          class="page-header-title-badge"
+          data-testid="page-header-title-badge"
+        >
+          <slot name="badge" />
+        </div>
+      </div>
+
+      <div
+        v-if="$slots.actions"
+        class="page-header-actions"
+        data-testid="page-header-actions"
+      >
+        <slot name="actions" />
+      </div>
+    </div>
+
+    <div
+      v-if="$slots.below"
+      class="page-header-section-below"
+      data-testid="page-header-section-below"
+    >
+      <slot name="below" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType } from 'vue'
+import type { BreadcrumbItem } from '@kong/kongponents'
+
+const props = defineProps({
+  title: {
+    type: String,
+    default: '',
+    required: true,
+  },
+  breadcrumbs: {
+    type: Array as PropType<BreadcrumbItem[]>,
+    default: () => ([]),
+  },
+})
+
+const hasBreadcrumbs = computed((): boolean => !!props.breadcrumbs.length)
```

Comment:
nit: you need to optionally chain here in case the user passes in `undefined`
```suggestion
const hasBreadcrumbs = computed((): boolean => !!props.breadcrumbs?.length)
```